src/Components/EntityState.cs

```diff
@@ -0,0 +1,8 @@
+﻿namespace Components
+{
+	public interface EntityState
+	{
+		bool TryAdd<Component>(Component component);
+		Component Get<Component>();
```

Comment:
@sblackler Surely it would be better for Get<Component> to just return null if the component didn't exist though? Not sure it would be a good idea to expect an exception over returning null?